includes/components/class-user.php

```diff
@@ -62,11 +62,46 @@ public function __construct( $args = [] ) {
 
 			// Alter templates.
 			add_filter( 'hivepress/v1/templates/site_footer_block', [ $this, 'alter_site_footer_block' ] );
+
+			// Update last seen.
+			add_action( 'wp_footer', [ $this, 'update_last_seen' ] );
 		}
 
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Updates user last seen.
+	 */
+	public function update_last_seen() {
+
+		// Get user ID.
+		$user_id = get_current_user_id();
+
+		// Get cache.
+		$last_seen = hivepress()->cache->get_user_cache( $user_id, 'last_seen' );
```

Comment:
It seems that we don't need cache if you added a user field, it already can be used as cache (also, user cache is implemented as user meta in the same way as custom fields).

Also move the 15min check to the condition instead of storage below, if I understand correctly if you check "last seen" right after it's refreshed, there will be a future time. You can save time() instead and check if < time()-15*60, so on the time of re-saving there will be the current timestamp instead of the future time.

Reply:
No, you're wrong, there will not be future time when you will check it.
It's saving future time to avoid duplicating if time() > last_seen + 15 * 60 conditions in each template file